src/main.rs

```diff
@@ -12,20 +12,104 @@ fn main() {
 
     // Parse command line arguments
     let args: Vec<String> = std::env::args().collect();
-    if args.len() < 3 {
+    if args.len() < 5 {
         eprintln!("Usage: {} <input wave filename> <output text filename>", args[0]);
         return
     }
 
     // Open the input wave file
     let mut reader = hound::WavReader::open(&args[1]).unwrap();
     let spec = reader.spec();
-    let channels = spec.channels;
+    let channels = spec.channels; // number of channels in input file
 
     // TODO: Modify this to process audio in blocks using your comb filter and write the result to an audio file.
-    //       Use the following block size:
+
+    // Define block size
     let block_size = 1024;
 
+    // Open output wave file
+    let mut writer = hound::WavWriter::create(&args[2], spec).unwrap();
+
+    // Assign sample rate from input
+    let sample_rate = spec.sample_rate as f32; 
+
+    // Command line comb filter argument (FIR or IIR)
+    let comb_filter_type = match args[3].as_str() { 
+        "FIR" => comb_filter::FilterType::FIR,
+        "IIR" => comb_filter::FilterType::IIR,
+        _ => {
+            eprintln!("Invalid filter type. Acceptable options as FIR or IIR.");
+            return;
+        }
+    };
+
+    // Command line filter parameter type (Gain or Delay)
+    let comb_filter_parameter = match args[4].as_str() {
+        "Gain" => comb_filter::FilterParam::Gain,
+        "Delay" => comb_filter::FilterParam::Delay,
+        _ => {
+            eprintln!("Invalid filter type. Acceptable options as Gain or Delay.");
+            return;
+        }
+    }; 
+
+    // Command line filter parameter argument 
+    let comb_filter_value: f32 = match args[5].parse() {
+        Ok(value) => value,
+        Err(_) => {
+            eprintln!("Invalid effect parameter value. Floating point number required.");
+            return;
+        }
+    }; 
+
+    // Set max_delay_sec
+    let mut max_delay = 0.0;
+
+    if let comb_filter::FilterParam::Delay = comb_filter_parameter {
+        max_delay = comb_filter_value;
+    }
+
+    // Instantiate CombFilter
+    let mut comb_filter = comb_filter::CombFilter::new(comb_filter_type, max_delay, sample_rate, channels as usize);
+    
```

Comment:
Should probably add a way to set the gain value. 
```suggestion

    // Instantiate CombFilter
    let mut comb_filter = comb_filter::CombFilter::new(comb_filter_type, max_delay, sample_rate, channels as usize);
    
   match comb_filter_parameter {
        comb_filter::FilterParam::Delay => comb_filter.set_param(comb_filter_parameter, comb_filter_value),
        comb_filter::FilterParam::Gain => comb_filter.set_param(comb_filter_parameter, comb_filter_value)
    };
```